### DISCUSSION TITLE
Communication between modules and DreamView

### DISCUSSION THREAD

**Discussion by chiragr15** [ApolloAuto/apollo#15149] [Q&A]
I wanted to know how the different modules(Routing,Localization,Planning) communicate the output to DreamView.

Is it through an API call? 
I understand that the output of the modules are packed into protocol buffers, but how are these sent to DreamView ?

**Comment by YuqiHuai** [comment]
I believe it is done through CyberRT. All modules publish their output to a channel in CyberRT, then Dreamview's backend looks at those messages and communicates with front-end through socket.

Output of the modules can be viewed using `cyber_monitor`.
